### ISSUE TITLE
microphone does nothing since lastest update 3.3 X64

### ISSUE THREAD

**Issue by DjSteveOhh** [telegramdesktop/tdesktop#17360]
### Steps to reproduce

my default recording settings in windows is stereomix as default device and my razer nari microphone as the default communication device.
in the previous version of telegram this worked perfect.
in the new version of telegrami have nari microphone as the default recording source , the meter shows me it's detecting sound but when i press record a message , silence is recorded , same goes for voice chats , other side can't hear me.
Work around for this bug , i have to change my windows recording settings to the nari microphone as being the default device. and set telegram recording to 'default'

### Expected behaviour

telegram should record from default communication device  instead of the default device..

### Actual behaviour

telegram only records from the default device ( windows )

### Operating system

Windows 10 21H1

### Version of Telegram Desktop

3.3 X64

### Installation source

Microsoft Store

**Comment by ilya-fedin** [comment]
tdesktop uses default device as default, but you can choose any device in settings. It's very unlikely tdesktop will start to use default communication device by default,

**Comment by DjSteveOhh** [comment]
in the previous version it worked perfect , since update it doesn't , only when i change my microphone as the default device telegram will record.

**Comment by ilya-fedin** [comment]
you don't need to change your default device in windows, you can choose the device in telegram settings

**Comment by DjSteveOhh** [comment]
when i go to call settings in telegram and have my microphone as default communication device i can see the meter doing it's thing , showing me it's detecting sound , BUT telegram records silence when making a voice recording / go in a voice chat.
When i changed my recording settings in windows with microphone as the default device  then telegram does record my voice.

**Comment by Aokromes** [comment]
mic works for me on windows 11 + telegram desktop 3.3 downloaded from webpage....

**Comment by ilya-fedin** [comment]
> When i changed my recording settings in windows with microphone as the default device then telegram does record my voice.

As far as I understand you, telegram uses not the device you want, why can't you just explicitly set the device in Telegram settings?

**Comment by DjSteveOhh** [comment]
weird thing is that when i have my normal settings ( mic = default ccommunication device ) and go to the call settings page in telegram the meter shows me there is sound from the mic , but it doesnt record ....

**Comment by ilya-fedin** [comment]
tdesktop has multiple audio libraries, apparently when you call another library is used. The legacy libtgvoip has its own audio backends, it's used when you call to devices with old version of Telegram. Call meter uses webrtc just like calls to new versions of Telegram IIRC, webrtc uses openal audio library like the rest of Telegram, it uses the common default device.

**Comment by ilya-fedin** [comment]
I see you don't answer, can this be closed?

**Comment by DjSteveOhh** [comment]
is it fixed then ?  where is new update ?

**Comment by Aokromes** [comment]
well, like i said, it works for me, unless someone else states the oposite the issue looks to be your configuration.

**Comment by ilya-fedin** [comment]
> is it fixed then ? where is new update ?

i don't see a bug. I said multiple times if you don't like the device tdesktop chooses by default, just choose it manually in tdesktop settings.

**Comment by Deebens** [comment]
I'm also having what I believe is the same, or at least similar, issue(s) - Telegram Desktop 3.3 refuses to respect my audio device settings, be it defaults set in windows or specific devices set in Telegram Desktop itself - The only way I can 'trick' it into using the correct devices is to outright disable all other devices before starting a call, which is impractical.

Time constraints got the better of me trying to diagnose the exact dimensions of the problem so I just launched a copy of 3.2.5 I had on another system, and the behaviour returned to normal, respecting my system defaults and making use of the correct devices.

**Comment by ilya-fedin** [comment]
@Aokromes can you close the issue since no one pointed what's the problem to manually choose the device in Telegram settings?

**Comment by Deebens** [comment]
The problem with manually choosing a device in Telegram settings, as I mentioned, is that the wrong device is used regardless of what my settings may be, either in windows or in Telegram Desktop itself - Whatever device I choose, the same (incorrect) input and output devices are used.

I can force the correct device for output by disabling all other audio output devices outright, but I was unable to get Telegram Desktop to use the correct microphone no matter what I did. I'd be happy to test further if there's specific information that would be helpful.

As I said, grabbing a copy of an older version of Telegram Desktop and running it on the same machine in the same configuration resulted in audio being routed correctly, so that alone should demonstrate that something has changed.

**Comment by ilya-fedin** [comment]
Sounds like a duplicate of #8435

**Comment by DjSteveOhh** [comment]
Mkay , i got it working again as i had it in the first place , first of all i updated from an older version to 3.3 , sound settings go to default only.
reinstalled 2.5.3 again , it worked again , updated to 3.3 , same issue ..
Uninstalled telegram , downloaded the 3.3 version from website , it's working as before  , stereomix is the default device and my headphones is the default communication device.
So if you upgrade from older version and you have sound troubles , uninstall telegram and download the 3.3 version and do a fresh install.
